contents/kotlin/NullSafety.md

```diff
@@ -0,0 +1,121 @@
+# Null Safety in Kotlin
+
+Author: [Pan Haozhe](https://github.com/Haozhe321)
+
+# Overview
+Another part of this book chapter gives an introduction to Kotlin. This piece aims to discuss the null safety feature in Kotlin, and what was previously missing in Java.
+
+# The Billion Dollar Mistake
+“I call it my billion-dollar mistake. It was the invention of the null reference…My goal was to ensure that all use of references should be absolutely safe, with **checking performed automatically by the compiler**. But I couldn't resist the temptation to put in a null reference, simply because it was so easy to implement.”  
```

Comment:
Can be put in a block quote (using `>`) instead. Also, although this quote is interesting, it is not immediately apparent how it adds value to your book chapter since you did not comment on this/ make any reference to this anywhere else in the book chapter (which is quite a waste) :/

Reply:
There are a few areas where this idea of "check done by compiler" is being expanded. 
1) In the section on nullable and non-nullable types
2) In the summary